hazelcast/src/main/java/com/hazelcast/util/Preconditions.java

```diff
@@ -148,12 +150,71 @@ public static int checkPositive(int value, String errorMessage) {
         return value;
     }
 
+    /**
+     * Tests if the newBackupCount count is valid.
+     *
+     * @param newBackupCount          the number of sync backups
+     * @param currentAsyncBackupCount the current number of async backups
+     * @return the newBackupCount
+     * @throws java.lang.IllegalArgumentException if newBackupCount is smaller than 0, or larger than the maximum
+     *                                            number of backups.
+     */
+    public static int checkBackupCount(int newBackupCount, int currentAsyncBackupCount) {
+        if (newBackupCount < 0) {
+            throw new IllegalArgumentException("backup-count can't be smaller than 0");
+        }
+
+        if (currentAsyncBackupCount < 0) {
+            throw new IllegalArgumentException("async-backup-count can't be smaller than 0");
+        }
+
+        if (newBackupCount > MAX_BACKUP_COUNT) {
+            throw new IllegalArgumentException("backup-count can't be larger than than " + MAX_BACKUP_COUNT);
```

Comment:
Good idea. Currently that isn't done in the existing code.

I would like to do it in one go for the whole preconditions library